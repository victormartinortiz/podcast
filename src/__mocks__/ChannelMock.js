const ChannelMock = [
  {
    id: 3709182,
    channel_id: 3709182,
    stream_id: 3709182,
    submission_style: 'none',
    writable: false,
    parent_channel_id: null,
    created_at: '2015-04-08T19:51:07.000Z',
    updated_at: '2020-03-03T19:38:12.000Z',
    title: 'Undisclosed',
    description:
      'The Undisclosed podcast investigates wrongful convictions, and the U.S. criminal justice system, by taking a closer look at the perpetration of a crime, its investigation, the trial, and ultimate verdict... and finding new evidence that never made it to court.',
    formatted_description:
      '<p>The Undisclosed podcast investigates wrongful convictions, and the U.S. criminal justice system, by taking a closer look at the perpetration of a crime, its investigation, the trial, and ultimate verdict... and finding new evidence that never made it to court.</p>',
    urls: {
      web_url: 'https://api.audioboom.com/channels/3709182',
      logo_image: {
        original: 'https://images.theabcdn.com/i/35369648'
      },
      banner_image: {
        original: 'https://images.theabcdn.com/i/35369643'
      }
    },
    category: {
      id: 265,
      title: 'News',
      description: null,
      updated_at: '2019-07-23T08:50:52.000Z',
      priority: 0
    },
    channel_clips_count: 349,
    total_plays: null
  }
];

export default ChannelMock;
