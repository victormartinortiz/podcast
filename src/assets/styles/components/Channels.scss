.channels {
  display: grid;
  grid-gap: 15px;
  padding: 15px;
  grid-template-columns: repeat(auto-fill, minmax(160px, 1fr));
}
a.channel {
  display: block;
  margin-bottom: 0.5em;
  color: #333;
  text-decoration: none;
}
.channel img {
  border-radius: 3px;
  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.15);
  width: 100%;
}
.subtitle {
  padding: 5px;
  font-size: 0.9em;
  font-weight: 600;
  margin: 0;
  text-align: center;
}
