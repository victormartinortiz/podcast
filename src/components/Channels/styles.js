/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';
import { MdPlayArrow } from 'react-icons/md';

export const Grid = styled.div`
  display: grid;
  grid-gap: 15px;
  padding: 15px;
  grid-template-columns: repeat(auto-fill, minmax(160px, 1fr));
`;

export const Channel = styled.a`
  position: relative;
  margin-bottom: 0.5em;
  text-decoration: none;
  color: #333;
  cursor: pointer;
`;

export const SubTitle = styled.h2`
  margin: 0;
  padding: 5px;
  text-align: center;
  font-size: 0.9em;
  font-weight: 600;
  color: #ffffff;
`;

export const PlayCircleOutline = styled(MdPlayArrow)`
  position: absolute;
  border-radius: 50px;
  font-size: 50px;
  color: #000;
  background-color: #ffffff;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  text-align: center;
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  transition: 0.3s ease;
  @media (min-width: 1024px) {
    ${Channel}:hover & {
      opacity: 1;
    }
  }
`;

export const Image = styled.img`
  width: 100%;
  border-radius: 3px;
  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.15);
  @media (min-width: 1024px) {
    ${Channel}:hover & {
      opacity: 30%;
    }
  }
`;
