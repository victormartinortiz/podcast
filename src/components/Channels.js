import React from 'react';
import Link from 'next/link';
import '../assets/styles/components/Channels.scss';

const Channels = ({ getChannels }) => (
    <div className="channels">
      {getChannels.map(channel => (
        <Link href={`/channel?id=${channel.id}`} key={channel.id}>
          <a className="channel">
            <img src={channel.urls.logo_image.original} alt="image" />
            <h2>{channel.title}</h2>
          </a>
        </Link>
      ))}
    </div>
  );

export default Channels;
