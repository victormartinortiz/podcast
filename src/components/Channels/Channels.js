import React from 'react';
import { Link } from '../../routes';
import slug from '../utils/slug';
import '../assets/styles/components/Channels.scss';

const Channels = ({ getChannels }) => (
  <div className="channels">
    {getChannels.map(channel => (
      <Link
        route='channel'
        params={{slug: slug(channel.title), id: channel.id }}
        key={channel.id}
      >
        <a className="channel">
          <img src={channel.urls.logo_image.original} alt="image" />
          <h2>{channel.title}</h2>
        </a>
      </Link>
    ))}
  </div>
);

export default Channels;
