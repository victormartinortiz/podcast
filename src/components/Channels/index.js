/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from '../../../routes';
import slug from '../../utils/slug';
import { Grid, Channel, Image, SubTitle } from './styles';

const Channels = ({ getChannels }) => (
  <Grid>
    {getChannels.map(channel => (
      <Link
        route='channel'
        params={{ slug: slug(channel.title), id: channel.id }}
        key={channel.id}
      >
        <Channel>
          <Image src={channel.urls.logo_image.original} alt='image' />
          <SubTitle>{channel.title}</SubTitle>
        </Channel>
      </Link>
    ))}
  </Grid>
);

export default Channels;
