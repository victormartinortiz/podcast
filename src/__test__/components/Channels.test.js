import React from 'react';
import { create } from 'react-test-renderer';
import Channels from '../../components/Channels';
import ChannelMock from '../../__mocks__/ChannelMock';

describe('<Channels />', () => {
  it('renders Channels unchanged', () => {
    const tree = create(
      <Channels getChannels={ChannelMock} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  })
});
