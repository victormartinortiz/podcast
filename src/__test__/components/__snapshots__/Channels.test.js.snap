// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Channels /> renders Channels unchanged 1`] = `
<div
  className="styles__Grid-sc-4r36jh-0 fhKoXs"
>
  <a
    className="styles__Channel-sc-4r36jh-1 jqIDMG"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      alt="image"
      className="styles__Image-sc-4r36jh-5 coSDSp"
      src="https://images.theabcdn.com/i/35369648"
    />
    <div
      className="styles__Overlay-sc-4r36jh-4 fEOFQn"
    >
      <svg
        className="styles__PlayCircleOutline-sc-4r36jh-3 hkaRrQ"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 5v14l11-7z"
        />
      </svg>
    </div>
    <h2
      className="styles__SubTitle-sc-4r36jh-2 imzgRX"
    >
      Undisclosed
    </h2>
  </a>
</div>
`;
